.github/actions/set-build-profile/action.yml

```diff
@@ -0,0 +1,22 @@
+name: Set Build Profile
+description: Set build profile to include debug checks for dev, alpha, and beta tags.
+runs:
+  using: "composite"
+  steps:
+    - shell: bash
+      run: |
+        TAG=${{ inputs.tag }}
+        LOWERTAG=$(echo "$TAG" | tr '[:upper:]' '[:lower:]')
+        if [[ "$LOWERTAG" == *dev* || "$LOWERTAG" == *alpha* || "$LOWERTAG" == *beta* ]]; then
+          BUILD_PROFILE=opt-test
+          IS_RELEASE=false
+        else
+          BUILD_PROFILE=release-dbgsymbols
```

Comment:
See below, this will have to branch based on whether it's windows or not. If windows, it needs to be `release`.